pkg/identity/utils.go

```diff
@@ -58,22 +58,36 @@ func writeFile(path string, dirmode, filemode os.FileMode, data []byte) error {
 	return nil
 }
 
-func statTLSFiles(certPath, keyPath string) TLSFilesStatus {
-	_, err := os.Stat(certPath)
-	hasCert := !os.IsNotExist(err)
+func statTLSFiles(certPath, keyPath string) (status TLSFilesStatus, err error) {
+	_, err = os.Stat(certPath)
+	hasCert := true
+	if err != nil {
+		if os.IsNotExist(err) {
+			hasCert = false
+		} else {
+			return NoCertNoKey, err
+		}
+	}
 
 	_, err = os.Stat(keyPath)
-	hasKey := !os.IsNotExist(err)
+	hasKey := true
```

Comment:
nit - maybe this should just be set once below line 61?

Reply:
done